debian/monsterz.6
.\" DO NOT MODIFY THIS FILE!  It was generated by help2man 1.36.
.TH MONSTERZ "6" "October 2007" "monsterz 0.7.0" "User Commands"
.SH NAME
monsterz \- a puzzle game of monsters
.SH SYNOPSIS
.B monsterz
[\fIOPTION\fR]...
.SH DESCRIPTION
Options
.TP
\fB\-h\fR, \fB\-\-help\fR
display this help and exit
.TP
\fB\-v\fR, \fB\-\-version\fR
display version information and exit
.TP
\fB\-f\fR, \fB\-\-fullscreen\fR
start in full screen mode
.TP
\fB\-m\fR, \fB\-\-nomusic\fR
disable music
.TP
\fB\-s\fR, \fB\-\-nosfx\fR
disable sound effects
.TP
\fB\-\-outfd\fR <fd>
output scores to file descriptor <fd>
.TP
\fB\-\-data\fR <dir>
set alternate data directory to <dir>
.HP
\fB\-\-score\fR <file> set score file to <file>
.SH AUTHOR
Written by Sam Hocevar, music by MenTaLguY, sound effects by Sun Microsystems,
Inc., Michael Speck, David White and the Battle for Wesnoth project, Mike
Kershaw and Sam Hocevar.
.SH "REPORTING BUGS"
Report bugs or suggestions to <sam@zoy.org>.
.SH COPYRIGHT
Copyright \(co 2005, 2006 Sam Hocevar <sam@zoy.org>
          (C) 1998 MenTaLguY <mental@rydia.net>
          (C) 2002, 2005 Sun Microsystems, Inc.
          (C) Michael Speck <kulkanie@gmx.net>
          (C) 2003 by David White <davidnwhite@optusnet.com.au> and the
              Battle for Wesnoth project
          (C) Mike Kershaw <dragorn@kismetwireless.net>
This program is free software; you can redistribute it and/or modify it under
the terms of the Do What The Fuck You Want To Public License, Version 2, as
published by Sam Hocevar. See http://sam.zoy.org/wtfpl/ for more details.
The sound effects are released under their own licences: applause.wav and
pop.wav are covered by the LGPL, the others are covered by the GPL.
.SH "SEE ALSO"
The in-game help option.


debian/changelog
monsterz (0.8.1-1) unstable; urgency=medium

  * Add gettext I18n
  * Add a full french translation

 -- Ferry Jérémie <jerem.ferry@gmail.com>  Sat, 22 Apr 2017 15:02:42 +0100

monsterz (0.7.1-9) unstable; urgency=low

  * Team upload.
  * Declare compliance with Debian Policy 3.9.6.
  * Vcs-Browser: Use https.
  * wrap-and-sort -sa.
  * Remove hardening-wrapper and use dpkg-buildflags instead. Fix Lintian error
    build-depends-on-obsolete-package. Add hardening.patch.
  * Add keywords to desktop file.

 -- Markus Koschany <apo@debian.org>  Sat, 07 Nov 2015 03:28:57 +0100

monsterz (0.7.1-8) unstable; urgency=medium

  * Team upload.
  * Do not use the obsolete and incorrect full path to dpkg-statoverride.
    Closes: #770057

 -- Peter Pentchev <roam@ringlet.net>  Sat, 29 Nov 2014 02:25:44 +0200

monsterz (0.7.1-7) unstable; urgency=low

  * Team upload.
  * debian/control:
    - Use canonical VCS-URI.
    - Add Breaks and Replaces relation to monsterz-data. This prevents that
      monsterz fails to upgrade. (Closes: #710733)
    - monsterz: Tighten dependency on monsterz-data (>= ${source:Version}).
  * Bump compatibility level to 9 and require debhelper >= 9 for automatic
    hardening build flags.
  * Use hardening-wrapper to harden the monsterz binary completely hence we can
    drop 040_makefile_cflags.patch.
  * Fix lintian issue copyright-refers-to-symlink-license and point to GPL-2
    and LGPL-2.1 licenses.
  * Add preinst script because dpkg does not replace symlinks with directories
    and vice versa, therefore maintainer scripts need to be used.

 -- Markus Koschany <apo@gambaru.de>  Mon, 01 Jul 2013 12:02:42 +0200

monsterz (0.7.1-6) unstable; urgency=low

  * Remove yes flag from --with-python2. (Closes: #710376).

 -- Barry deFreese <bdefreese@debian.org>  Thu, 30 May 2013 05:40:28 -0400

monsterz (0.7.1-5) unstable; urgency=low

  * Add startup animation patch. (Closes: #439807).
    - Thanks to A Mennucc for the patch.
  * Move existing manpage to section 6. (Closes: #605981).
  * Change packaging to dh7 style.
  * Add patch to pass CFLAGS to Makefile.
    - Add hardening flags.
  * Change to source format 3.0 quilt.
    - Drop quilt build-dep.
  * Bump Standards Version to 3.9.4.

 -- Barry deFreese <bdefreese@debian.org>  Sun, 26 May 2013 21:35:56 -0400

monsterz (0.7.1-4) unstable; urgency=low

  * Add monsterz.postrm to purge /var/games/monsterz  
  * Add myself in Uploaders
  * Standards-Version: 3.8.2, no change 

 -- Gonéri Le Bouder <goneri@rulezlan.org>  Mon, 27 Jul 2009 18:42:39 +0200

monsterz (0.7.1-3) unstable; urgency=low

  [ Barry deFreese ]
  * Add myself to uploaders.
  * 020_fix_blit_crash.diff - (Closes: #523877). 
    + Fix crash with blit in pygame 1.8.1.
    + Thanks to Arnaud Jeansen and RafaÅ‚ Psota.
  * Add set -e to postinst.
  * Fix up manpage what-is section.
  * Bump Standards Version to 3.8.1.

 -- Barry deFreese <bdefreese@debian.org>  Fri, 15 May 2009 21:18:05 -0400

monsterz (0.7.1-2) unstable; urgency=low

  * Remove deprecated Encoding key from .desktop file

  [ Sam Hocevar ]
  * debian/control:
    + Use quilt for patch management.

  * debian/patches/010_64-bit-alignment-issues-with-python2.5.diff:
    + New patch from upstream. Fix a startup crash on 64-bit systems
      (Closes: #373905).

 -- Sam Hocevar (Debian packages) <sam+deb@zoy.org>  Fri, 09 May 2008 16:33:52 +0000

monsterz (0.7.1-1) unstable; urgency=high

  [ Eddy Petrișor ]
  * removed invalid category "Application" from the desktop file

  [ Cyril Brulebois ]
  * Adjusted XS-Vcs-Svn and XS-Vcs-Browser fields in the control file.

  [ Barry deFreese ]
  * Add simple man page
  * Remove short description from long description
  * Make distclean not ignore errors

  [ Sam Hocevar ]
  * New upstream release. Removes potentially offensive language from the
    title page (Closes: #456781).
  * debian/control:
    + Set policy to 3.7.3.
    + Added Homepage: field.
    + Fixed Vcs: fields, they don't need "XS-" prepended.
  * debian/copyright:
    + Add link to the GPL and the LGPL texts.

 -- Sam Hocevar (Debian packages) <sam+deb@zoy.org>  Wed, 11 Jul 2007 11:06:35 +0300

monsterz (0.7.0-2) unstable; urgency=low

  * Moved packaging to the Debian Games Team.
  * debian/control:
    + Added Vcs fields.
  * debian/compat:
    + Bumped level to 5.
  * debian/monsterz.desktop:
    + Added desktop file, courtesy of Jean Privat (Closes: #395908).

 -- Sam Hocevar (Debian packages) <sam+deb@zoy.org>  Fri, 30 Mar 2007 12:07:13 +0200

monsterz (0.7.0-1) unstable; urgency=low

  * Initial Debian upload (Closes: #329323).

 -- Sam Hocevar (Debian packages) <sam+deb@zoy.org>  Wed, 10 May 2006 01:11:29 -0500



monsterz.c
#include <sys/types.h>
#include <stdio.h>
#include <unistd.h>
#include <stdlib.h>

/*
 * monsterz.c: setgid wrapper for monsterz.py
 * $Id: monsterz.c 76 2005-03-09 14:10:06Z sam $
 *
 * Copyright: (c) 2005 Sam Hocevar <sam@zoy.org>
 *   This program is free software; you can redistribute it and/or
 *   modify it under the terms of the Do What The Fuck You Want To
 *   Public License, Version 2, as published by Sam Hocevar. See
 *   http://sam.zoy.org/projects/COPYING.WTFPL for more details.
 */

int main(int argc, char **argv)
{
    char buffer[BUFSIZ];
    char *data = DATADIR;
    char *script = DATADIR "/monsterz.py";
    char *score = SCOREFILE;
    char *outfd = "42";
    char **newargv;
    int ret, pos = 0, gid, egid, fd[2];

    egid = getegid();
    gid = getgid();
    pipe(fd);

    /* Spawn a child, drop privileges and run our script in the father */
    switch(fork())
    {
        case 0: /* child */
            close(fd[1]);
            break;
        case -1: /* error */
            return -1;
        default: /* father */
            close(fd[0]);
            dup2(fd[1], atoi(outfd));
            /* drop privileges */
            if(egid != gid)
                setegid(gid);
            /* build a new argv */
            newargv = malloc((argc + 7) * sizeof(char *));
            for(pos = 1; pos < argc; pos++)
                newargv[pos] = argv[pos];
            newargv[0] = script;
            newargv[argc++] = "--outfd";
            newargv[argc++] = outfd;
            newargv[argc++] = "--data";
            newargv[argc++] = data;
            newargv[argc++] = "--score";
            newargv[argc++] = score;
            newargv[argc] = NULL;
            /* run our script */
            execv(script, newargv);
            fprintf(stderr, "%s: could not find `%s'.\n", argv[0], script);
            return -1;
    }

    /* Handle our child’s messages */
    for(;;)
    {
        ret = read(fd[0], buffer + pos, 1);
        if(ret <= 0)
            break; /* EOF or error */

        if(buffer[pos] == '\n' && buffer[pos - 1] == '\n')
        {
            FILE *f = fopen(score, "w");
            buffer[pos] = '\0'; pos = 0;
            if(f)
            {
                fprintf(f, "%s", buffer);
                fclose(f);
            }
            else
                fprintf(stderr, "%s: unable to write score file `%s'\n",
                        argv[0], score);
        }
        else if(pos++ >= BUFSIZ - 8)
            return -1; /* The script is doing nasty stuff... quit */
    }

    return 0;
}

